pinot-core/src/main/java/org/apache/pinot/core/query/executor/ServerQueryExecutorV1Impl.java

```diff
@@ -94,7 +96,8 @@ public synchronized void shutDown() {
   }
 
   @Override
-  public DataTable processQuery(ServerQueryRequest queryRequest, ExecutorService executorService) {
+  public DataTable processQuery(ServerQueryRequest queryRequest, ExecutorService executorService,
+      @Nullable StreamObserver<Server.ServerResponse> responseObserver) {
```

Comment:
I think it is possible to have a separate implementation. Gradually over time, this might need more changes we wiull find ourselves polluting this method for streaming v/s non-streaming if branches.